From 2db9081b379614ecc19331634d31ca827b9cf9e6 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Tue, 27 Dec 2016 18:53:40 +0100
Subject: Fix arguments when running .desktop files with runservice

Without this patch the command run with runservice has a duplicate first argument
as QProcess already prepends the executable name.
---
 libdiscover/backends/PackageKitBackend/runservice/main.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libdiscover/backends/PackageKitBackend/runservice/main.cpp b/libdiscover/backends/PackageKitBackend/runservice/main.cpp
index b56917a..693d80c 100644
--- a/libdiscover/backends/PackageKitBackend/runservice/main.cpp
+++ b/libdiscover/backends/PackageKitBackend/runservice/main.cpp
@@ -39,5 +39,7 @@ int main(int argc, char** argv)
     QTextStream cerr(stderr);
     KIO::DesktopExecParser execParser(_service, {});
 
-    return !QProcess::startDetached(KIO::DesktopExecParser::executableName(_service.exec()), execParser.resultingArguments());
+    auto args = execParser.resultingArguments();
+    const auto execName = args.takeFirst();
+    return !QProcess::startDetached(execName, args);
 }
-- 
cgit v0.11.2

